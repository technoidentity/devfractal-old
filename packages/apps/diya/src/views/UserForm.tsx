import React from 'react'
import {
  Box,
  Button,
  Column,
  Columns,
  formComponent,
  Image,
  Media,
  MediaContent,
  Radio,
  required,
  Section,
  Simple,
  Title,
} from 'technoidentity-devfractal'
import { User } from '../common'
import { HeadTitle } from '../components'

// const validationSchema = yup.object().shape({
//   name: yup.string().required(),
//   driverId: yup.string().required(),
//   phone: yup.string().required(),
//   driverLicence: yup.string().required(),
//   adharNumber: yup.string().required(),
// })

export const UserForm = formComponent(User, ({ initial, edit, onSubmit }) => (
  <>
    <Section>
      <HeadTitle>{edit ? 'Edit' : 'Create'} User</HeadTitle>
    </Section>
    <Section>
      <Simple.Form initialValues={initial} onSubmit={onSubmit}>
        <Columns>
          <Column>
            <Title size="5" textColor="info">
              Personal Details
            </Title>
            <Simple.Text name="name" validations={[required()]} />
            {/* <Simple.Text
                name="driverID"
                label="Driver ID"
                validations={[required()]}
              /> */}
            <Simple.Telephone name="phone" validations={[required()]} />
            <Simple.Text
              name="licenceNo"
              label="Licence No."
              validations={[required()]}
            />
            <Simple.Text name="adharNumber" validations={[required()]} />

            <Simple.Select name="role" fullWidth>
              <option value="Admin">Admin</option>
              <option value="Reporter">Reporter</option>
              <option value="Dispatcher">Dispatcher</option>
            </Simple.Select>
            <Simple.Text name="address1" />
            <Simple.Text name="address2" />
          </Column>

          <Column>
            <Title size="5" textColor="info">
              Bank Details
            </Title>
            <Simple.Text name="accountName" validations={[required()]} />
            <Simple.Text name="accountNumber" validations={[required()]} />

            <Simple.Text
              name="confirmAccountNumber"
              validations={[required()]}
            />
            <Simple.Text name="bankName" validations={[required()]} />
            <Simple.Text name="bankBranch" validations={[required()]} />
            <Simple.Text
              name="branchIFSCNumber"
              label="Branch IFSC Number"
              validations={[required()]}
            />
            <Simple.RadioGroup name="verified" defaultValue="pending">
              <Radio value="yes"> yes</Radio>
              <Radio value="pending"> In process</Radio>
            </Simple.RadioGroup>
          </Column>

          <Column>
            <Title size="6" textColor="info">
              Profile Photo
            </Title>

            <Box>
              <Media>
                <MediaContent>
                  <Image
                    size="128x128"
                    src="https://bulma.io/images/placeholders/128x128.png"
                  />
                </MediaContent>
              </Media>
            </Box>

            <Button variant="dark">Upload Photo</Button>
            <div style={{ paddingTop: '10px' }}>
              <Simple.Text name="emergencyContactPerson" />
              <Simple.Text name="emergencyContactNumber" />
              <Simple.Select name="relation" fullWidth>
                <option value="father">Father</option>
                <option value="mother">Mother</option>
                <option value="spouse">Spouse</option>
              </Simple.Select>
            </div>
          </Column>
        </Columns>

        <Simple.FormButtons />
      </Simple.Form>
    </Section>
  </>
))
