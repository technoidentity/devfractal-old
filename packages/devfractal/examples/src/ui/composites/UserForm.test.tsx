import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import { UserForm } from './UserForm'

test('User Form Example', () => {
  const component: ReactTestRenderer = renderer.create(<UserForm />)
  expect(component.toJSON()).toMatchSnapshot()
})
