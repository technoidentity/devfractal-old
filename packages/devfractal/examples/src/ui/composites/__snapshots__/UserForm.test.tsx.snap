// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Form Example 1`] = `
<form
  action="#"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="name"
    >
      Name:
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        id="name"
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="username"
    >
      UserName:
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        id="username"
        name="username"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="email"
    >
      Email
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="phone"
    >
      Phone:
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        id="phone"
        name="phone"
        onBlur={[Function]}
        onChange={[Function]}
        type="tel"
        value={0}
      />
    </div>
  </div>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="website"
    >
      Website:
    </label>
    <div
      className="control"
    >
      <input
        className="input"
        id="website"
        name="website"
        onBlur={[Function]}
        onChange={[Function]}
        type="url"
        value=""
      />
    </div>
  </div>
  <div
    className="buttons"
  >
    <button
      className="button is-info"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
    <button
      className="button is-danger"
      disabled={true}
      onClick={[Function]}
      type="reset"
    >
      Reset
    </button>
  </div>
  <pre>
    {
  "name": "",
  "username": "",
  "email": "",
  "phone": 0,
  "website": ""
}
  </pre>
</form>
`;
